go/enclave/rollupchain/rollup_chain.go

```diff
@@ -577,50 +577,57 @@ func (rc *RollupChain) processState(batch *core.Batch, txs []*common.L2Tx, state
 	return rootHash, executedTransactions, txReceipts, depositReceipts
 }
 
-func (rc *RollupChain) isValidBatch(batch *core.Batch, rootHash common.L2RootHash, txReceipts []*types.Receipt, depositReceipts []*types.Receipt, stateDB *state.StateDB) bool {
+// Checks the internal validity of the batch.
+func (rc *RollupChain) isInternallyValidBatch(batch *core.Batch) (types.Receipts, error) {
+	stateDB, err := rc.storage.CreateStateDB(batch.Header.ParentHash)
+	if err != nil {
+		return nil, fmt.Errorf("could not create stateDB. Cause: %w", err)
+	}
+
+	// calculate the state to compare with what is in the batch
+	rootHash, successfulTxs, txReceipts, depositReceipts := rc.processState(batch, batch.Transactions, stateDB)
+	if len(successfulTxs) != len(batch.Transactions) {
+		return nil, fmt.Errorf("all transactions that are included in a batch must be executed")
```

Comment:
do we filter out failed transaction receipts from batches? Don't validators need to know them?

Reply:
Don't know. I was just shuffling this code around. Should work just like it did in the rollup case.

Reply:
yeah not a blocker for this PR, but an interesting question I think

Reply:
not sure that error message is legit, for example

Reply:
ah, they're called "executedTransactions" in the processState() function, that var should probably change